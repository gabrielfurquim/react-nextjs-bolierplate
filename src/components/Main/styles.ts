import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  height: 100%;
  width: 100%;
  padding: 3rem;
  text-align: center;
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`

export const Ilustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`
export const List = styled.ul`
  list-style: none;
`

export const ListItem = styled.li`
  margin: 1rem 0;
  letter-spacing: 1.4px;
`
