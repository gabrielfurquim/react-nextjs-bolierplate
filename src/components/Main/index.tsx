import * as S from './styles'

type Rocket = {
  id: string
}

type Props = {
  title: string
  description: string
  rockets: [Rocket]
}

function Main({
  title = 'SpaceX Rockets',
  description = 'Get space-x rockets using graphql public api.',
  rockets
}: Props) {
  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <S.List>
        {rockets.map((rocket) => (
          <S.ListItem key={rocket.id}>{rocket.id}</S.ListItem>
        ))}
      </S.List>
      <S.Ilustration
        src="/img/rocket.svg"
        alt="Um foguete em fase de lanÃ§amento."
      ></S.Ilustration>
    </S.Wrapper>
  )
}
export default Main
