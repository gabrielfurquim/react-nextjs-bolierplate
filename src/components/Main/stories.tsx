import { withKnobs, text } from '@storybook/addon-knobs'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
}

export const Basic = () => (
  <Main
    title={text('Title', 'SpaceX Rockets')}
    description={text(
      'Description',
      'Get space-x rockets using graphql public api.'
    )}
    rockets={[{ id: 'rocket' }]}
  />
)
