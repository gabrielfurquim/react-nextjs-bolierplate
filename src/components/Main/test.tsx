import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main rockets={[{ id: 'rocket' }]} />)

    expect(
      screen.getByRole('heading', { name: /SpaceX Rockets/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main rockets={[{ id: 'rocket' }]} />)

    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b'
    })
  })
})
