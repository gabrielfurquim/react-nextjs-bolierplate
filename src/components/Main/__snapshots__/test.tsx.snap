// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  height: 100%;
  width: 100%;
  padding: 3rem;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 2.5rem;
}

.c2 {
  font-size: 2rem;
  font-weight: 400;
}

.c5 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

.c3 {
  list-style: none;
}

.c4 {
  margin: 1rem 0;
  -webkit-letter-spacing: 1.4px;
  -moz-letter-spacing: 1.4px;
  -ms-letter-spacing: 1.4px;
  letter-spacing: 1.4px;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    SpaceX Rockets
  </h1>
  <h2
    class="c2"
  >
    Get space-x rockets using graphql public api.
  </h2>
  <ul
    class="c3"
  >
    <li
      class="c4"
    >
      rocket
    </li>
  </ul>
  <img
    alt="Um foguete em fase de lanÃ§amento."
    class="c5"
    src="/img/rocket.svg"
  />
</main>
`
